### PR TITLE
More translations

### PR DESCRIPTION
**Pull request by ingodahn**
More translations in actions.php, course/addassessment2form.php and header.php. These are all php localizations.

git reports also differences of my and your master 
assess2/vue-src/package-lock.json
assess2/vue-src/src/i18n.js
assess2/vue/js/special.js.map

I have not included those versions in feature_de because I am afraid they might spoil the English version. They shall go into feature_vcrp

### CODE REVIEW

File: actions.php [hunk 1]
```diff
@@ -274,32 +274,31 @@
 				$stm = $DBH->prepare($query);
 				$stm->execute(array(':code'=>$code, ':id'=>$id));
 
-				$message  = "<h3>This is an automated message from $installname.  Do not respond to this email</h3>\r\n";
-				$message .= "<p>Your username was entered in the Reset Password page.  If you did not do this, you may ignore and delete this message. ";
-				$message .= "If you did request a password reset, click the link below, or copy and paste it into your browser's address bar.  You ";
-				$message .= "will then be prompted to choose a new password.</p>";
+				$message  = "<h3>".sprintf('This is an automated message from %s. Do not respond to this email',$installname)."</h3>\r\n";
```
Comment (drlippman) [inline actions.php]: This line is missing the _()

File: admin/forms.php [hunk 2]
```diff
@@ -770,21 +770,21 @@ function courselevelchg(el) {
 		echo '</span><br class=form />';
 
 		if (!isset($CFG['CPS']['deftime']) || $CFG['CPS']['deftime'][1]==1) {
-			echo "<span class=form>Default start/end time for new items:</span><span class=formright>";
+			echo "<span class=form>",_("Default start/end time for new items"),":</span><span class=formright>";
 			echo 'Start: <input name="defstime" type="text" size="8" value="'.Sanitize::encodeStringForDisplay($defstimedisp).'"/>, ';
```
Comment (drlippman) [inline admin/forms.php]: Did you want to add translation to this line and the next?

File: admin/forms.php [hunk 3]
```diff
@@ -770,21 +770,21 @@ function courselevelchg(el) {
...
 			echo 'end: <input name="deftime" type="text" size="8" value="'.Sanitize::encodeStringForDisplay($deftimedisp).'"/>';
 			echo '</span><br class="form"/>';
 		}
 
 		if (!isset($CFG['CPS']['unenroll']) || $CFG['CPS']['unenroll'][1]==1) {
-			echo "<span class=form>Allow students to self-<u>un</u>enroll</span><span class=formright>";
+			echo "<span class=form>",_("Allow students to self-<u>un</u>enroll"),"</span><span class=formright>";
 			echo '<input type=radio name="allowunenroll" value="0" ';
 			if (($allowunenroll&1)==0) { echo "checked=1";}
 			echo '/> No <input type=radio name="allowunenroll" value="1" ';
```
Comment (drlippman) [inline admin/forms.php]: Add translation here and below?

File: admin/forms.php [hunk 4]
```diff
@@ -813,7 +813,7 @@ function courselevelchg(el) {
 			echo '</div>';
 			echo '<div class="blockitems hidden">';
 
-			echo '<p>View <a target="_blank" href="../help.php?section=lti">integration setup instructions</a></p>';
+			echo '<p>',_('View <a target="_blank" href="../help.php?section=lti">integration setup instructions'),'</a></p>';
```
Comment (drlippman) [inline admin/forms.php]: Do you want all the <a> inside the translation string?

File: admin/forms.php [hunk 5]
```diff
@@ -770,21 +770,21 @@ function courselevelchg(el) {
...
-			echo "<span class=form>Allow students to self-<u>un</u>enroll</span><span class=formright>";
+			echo "<span class=form>",_("Allow students to self-<u>un</u>enroll"),"</span><span class=formright>";
 			echo '<input type=radio name="allowunenroll" value="0" ';
 			if (($allowunenroll&1)==0) { echo "checked=1";}
 			echo '/> No <input type=radio name="allowunenroll" value="1" ';
 			if (($allowunenroll&1)==1) { echo "checked=1";}
 			echo '/> Yes </span><br class=form />';
 
-			echo "<span class=form>Allow students to self-enroll</span><span class=formright>";
+			echo "<span class=form>",_("Allow studentsAllow students to self-enroll"),"</span><span class=formright>";
```
Comment (drlippman) [inline admin/forms.php]: Looks like a typo with duplicated string here

**Comment by ingodahn** [comment]
I translated just what I needed for my planned screencast, will continue to
work on that. You may delay merging.
Ingo

Am Sa., 22. Feb. 2020 um 18:58 Uhr schrieb David Lippman <
notifications@github.com>:

**Comment by ingodahn** [comment]
I'll comment inline Sometimes it seems I have to restart the server to see
the translations working. Will check these are in.

Am Sa., 22. Feb. 2020 um 18:58 Uhr schrieb David Lippman <
notifications@github.com>:

> *@drlippman* requested changes on this pull request.
>
> Looks good, but I spotted a few issues.
> ------------------------------
>
> In actions.php
> <https://github.com/drlippman/IMathAS/pull/159#discussion_r382929584>:
>
> > @@ -274,32 +274,31 @@
>  				$stm = $DBH->prepare($query);
>  				$stm->execute(array(':code'=>$code, ':id'=>$id));
>
> -				$message  = "<h3>This is an automated message from $installname.  Do not respond to this email</h3>\r\n";
> -				$message .= "<p>Your username was entered in the Reset Password page.  If you did not do this, you may ignore and delete this message. ";
> -				$message .= "If you did request a password reset, click the link below, or copy and paste it into your browser's address bar.  You ";
> -				$message .= "will then be prompted to choose a new password.</p>";
> +				$message  = "<h3>".sprintf('This is an automated message from %s. Do not respond to this email',$installname)."</h3>\r\n";
>
> This line is missing the _()
>
Will correct

> ------------------------------
>
> In admin/forms.php
> <https://github.com/drlippman/IMathAS/pull/159#discussion_r382929764>:
>
> > @@ -770,21 +770,21 @@ function courselevelchg(el) {
>  		echo '</span><br class=form />';
>
>  		if (!isset($CFG['CPS']['deftime']) || $CFG['CPS']['deftime'][1]==1) {
> -			echo "<span class=form>Default start/end time for new items:</span><span class=formright>";
> +			echo "<span class=form>",_("Default start/end time for new items"),":</span><span class=formright>";
>  			echo 'Start: <input name="defstime" type="text" size="8" value="'.Sanitize::encodeStringForDisplay($defstimedisp).'"/>, ';
>
> Did you want to add translation to this line and the next?
>
Should have been translated

> ------------------------------
>
> In admin/forms.php
> <https://github.com/drlippman/IMathAS/pull/159#discussion_r382929782>:
>
> >  			echo 'Start: <input name="defstime" type="text" size="8" value="'.Sanitize::encodeStringForDisplay($defstimedisp).'"/>, ';
>  			echo 'end: <input name="deftime" type="text" size="8" value="'.Sanitize::encodeStringForDisplay($deftimedisp).'"/>';
>  			echo '</span><br class="form"/>';
>  		}
>
>  		if (!isset($CFG['CPS']['unenroll']) || $CFG['CPS']['unenroll'][1]==1) {
> -			echo "<span class=form>Allow students to self-<u>un</u>enroll</span><span class=formright>";
> +			echo "<span class=form>",_("Allow students to self-<u>un</u>enroll"),"</span><span class=formright>";
>  			echo '<input type=radio name="allowunenroll" value="0" ';
>  			if (($allowunenroll&1)==0) { echo "checked=1";}
>  			echo '/> No <input type=radio name="allowunenroll" value="1" ';
>
> Add translation here and below?
>
Should have been translated

**Comment by ingodahn** [comment]
Revised, please re-check

**Comment by drlippman** [comment]
Merged manually.

**Pull request closed without merging**